### PR TITLE
feat: add dataset `PATCH` endpoint

### LINKED ISSUE #3396: [FEATURE] Add `PATCH` endpoint for FeedbackDatasets
**Is your feature request related to a problem? Please describe.**
Yes, some attributes of the `FeedbackDataset` cannot be updated like the `name` or `guidelines`.

**Describe the solution you'd like**
Add a new `PATCH /api/v1/datasets/:dataset_id` endpoint that allows to update the following attributes:
- `name`
- `guidelines`

### PR DESCRIPTION
**Pull request by gabrielmbmb**
# Description

This PR adds a new endpoint `PATCH /api/v1/datasets/{dataset_id}` that allows to partially update a dataset. The attributes that can be updated are `name` and `guidelines`.

Closes #3396

**Type of change**

- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

Manually in a local deployment and I've added unit tests covering this new endpoint.

**Checklist**

- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [x] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

**Review by alvarobartt — APPROVED** [review]
LGTM! Are we allowing updates on both ready and draft datasets, right? If not, shouldn't we add a check making sure the status is still draft? Thanks in advance!

**Comment by gabrielmbmb** [comment]
@alvarobartt yes, the idea is to allow to update these attributes on both status!

**Review by frascuchon — APPROVED** [review]
👍  I've just added the changelog entry for this and a new tests check dataset update with `name=None` and `guidelines=None`

File: src/argilla/server/schemas/v1/datasets.py [hunk 1]
```diff
@@ -89,17 +97,26 @@ class Datasets(BaseModel):
...
+    constr(
+        min_length=DATASET_CREATE_GUIDELINES_MIN_LENGTH,
+        max_length=DATASET_CREATE_GUIDELINES_MAX_LENGTH,
+    ),
+    Field(..., description="Dataset guidelines"),
+]
+
+
 class DatasetCreate(BaseModel):
     name: str
```
Comment (frascuchon) [inline src/argilla/server/schemas/v1/datasets.py]: I realized that we should add a min/max bound for the dataset name

Comment (gabrielmbmb) [reply src/argilla/server/schemas/v1/datasets.py]: Done!

**Review by frascuchon — APPROVED** [review]

**Pull request merged**